src/guest_memory.rs

```diff
@@ -236,6 +237,68 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     }
 }
 
+struct GuestMemoryRange<'a, R> {
+    region: &'a R,
+    start: MemoryRegionAddress,
+    len: usize,
+}
+
+struct GuestMemoryRangeIter<'a, M: GuestMemory + ?Sized> {
+    memory: &'a M,
+    addr: GuestAddress,
+    count: usize,
+    total: usize,
+
+    // Stores the length of the current GuestMemoryRange to
+    // check for bugs.
+    _last_length: Cell<usize>,
+}
+
+impl<'a, M: GuestMemory + ?Sized> GuestMemoryRangeIter<'a, M> {
```

Comment:
Can you add comments to the public interface (pls pls pls)? Also some comments about what count and total are would be nice, it's not straightforward when reading the code the first time.

Reply:
Done, in fact I made the whole of `GuestMemoryRangeIter` part of the public interface of `vm_memory`.

`total` is now named `so_far`.